client/state/sites/selectors.js

```diff
@@ -87,6 +88,30 @@ export function isJetpackSite( state, siteId ) {
 }
 
 /**
+ * Returns true if the Jetpack site is running a version meeting the specified
+ * minimum, or false if the Jetpack site is running an older version. Returns
+ * null if the version cannot be determined or if not a Jetpack site.
+ *
+ * @param  {Object}   state   Global state tree
+ * @param  {Number}   siteId  Site ID
+ * @param  {String}   version Minimum version
+ * @return {?Boolean}         Whether running minimum version
+ */
+export function isJetpackMinimumVersion( state, siteId, version ) {
+	const isJetpack = isJetpackSite( state, siteId );
+	if ( ! isJetpack ) {
+		return null;
+	}
+
+	const siteVersion = getSiteOption( state, siteId, 'jetpack_version' );
+	if ( ! siteVersion ) {
+		return null;
```

Comment:
>won't returning `null` here cause a failure down the line in `EditorDrawerTaxonomies` since you are explicitly checking for a value of `false`

I think at worst it would result in a 404 error on the request, and this could only happen if visiting the editor with a non-primed sites state. The alternative would be to hide the taxonomy accordions for all sites until after we can verify that the site is either not Jetpack or meets the satisfying version.